### PR TITLE
Fix character sort

### PR DESCRIPTION
**Pull request by PhilPJL**
Do we need this to be controlled by a flag?

### CODE REVIEW

File: SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    [HarmonyPatch(typeof(CharacterSelectionModal), "CharactersFiltered")]
+    [SuppressMessage("Minor Code Smell", "S101:Types should be named in PascalCase", Justification = "Patch")]
+    internal static class CharacterSelectionModal_CharactersFiltered
+    {
+        public static bool Prefix(List<RulesetCharacterHero.Snapshot> filteredHeroesList, List<CharacterPlateToggle> ___characterPlates)
+        {
+            // Duplicate the logic of CharactersPanel.CharactersFiltered to fix mapping of filteredHeroesList to character plates
+
+            var cpDict = ___characterPlates.ToDictionary(cp => Path.GetFileNameWithoutExtension(cp.Filename), cp => cp);
+            var numHeros = filteredHeroesList.Count;
```
Comment (ThyWoof) [inline SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs:18]: In case you wanna fix en typo

**Review by ThyWoof — DISMISSED** [review]

File: SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs (hunk 1 above)
Comment (PhilPJL) [reply SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs:18]: which typo?  numHeros should be numHeroes?

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs:18]: yep but EN is my second language ;-)

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs:18]: there is a hidden sorting settings flag I'm using to protect them. Flag always nice to make it simple for us to search the code by setting name.

Comment (ThyWoof) [reply SolastaCommunityExpansion/Patches/GameUiCharactersPanel/CharacterSelectionModalPatcher.cs:18]: FutureFeatureSorting

**Review by ThyWoof — APPROVED** [review]

**Pull request merged**